src/mpi/romio/adio/ad_gpfs/ad_gpfs_aggrs.c

```diff
@@ -769,3 +793,82 @@ void ADIOI_GPFS_Calc_others_req(ADIO_File fd, int count_my_req_procs,
 #endif
     TRACE_ERR("Leaving ADIOI_GPFS_Calc_others_req\n");
 }
+
+/*
+ *  Alltoall with MPI_Aint for sdisps/rdisps
+ *
+ *  If the disps are actually small enough to fit in an int, it
+ *  creates int arrays and calls regular MPI_Alltoall.
+ *  Otherwise it does a bunch of copies to make the data contiguous
+ *  so it can fit in int displacements.
+ */
+static int
+MY_Alltoallv(void *sbuf, int *scounts, MPI_Aint * sdisps, MPI_Datatype stype,
+             void *rbuf, int *rcounts, MPI_Aint * rdisps, MPI_Datatype rtype, MPI_Comm comm)
+{
+    int sizeof_stype, sizeof_rtype;
+    int rv, i;
+    int nranks;
+    int disps_are_small_enough;
+    int *sdisps_int;
+    int *rdisps_int;
+
+    MPI_Comm_size(comm, &nranks);
+    MPI_Type_size(stype, &sizeof_stype);
+    MPI_Type_size(rtype, &sizeof_rtype);
+    sdisps_int = ADIOI_Malloc(2 * nranks * sizeof(int));
+    rdisps_int = &sdisps_int[nranks];
+
+    disps_are_small_enough = 1;
+    for (i = 0; i < nranks && disps_are_small_enough; ++i) {
+        if (sdisps[i] != (int) sdisps[i]) {
+            disps_are_small_enough = 0;
+        }
+    }
+    for (i = 0; i < nranks && disps_are_small_enough; ++i) {
+        if (rdisps[i] != (int) sdisps[i]) {
+            disps_are_small_enough = 0;
+        }
+    }
+
+    if (disps_are_small_enough) {
+        for (i = 0; i < nranks; ++i) {
+            sdisps_int[i] = sdisps[i];
+        }
+        for (i = 0; i < nranks; ++i) {
+            rdisps_int[i] = rdisps[i];
+        }
+        rv = MPI_Alltoallv(sbuf, scounts, sdisps_int, stype,
+                           rbuf, rcounts, rdisps_int, rtype, comm);
+        ADIOI_Free(sdisps_int);
+        return rv;
+    }
+
```

Comment:
How about wrap the following code into the `else` branch?

Reply:
okay, I'm making that change